Reduce backend load created from session trackers

rosstimothy opened the pull request:
Session trackers were originally added to facilitate joining sessions and enforcing moderation policies. When a session is created, a new tracker is written to the backend and a background routine is spawned to periodically update the status of the tracker until the session is terminated. This can cause a massive amount of backend activity for a cluster that is spawning large quantities of sessions per second. While in most cases where humans are starting the sessions this isn't a problem, any machine id heavy use cases could trigger backend throttling. Since non-interactive sessions and sessions started by tbot are not meant to be joined or moderated, the existence of a session tracker for them doesn't provide much benefit, especially now that session recordings are disabled for non-interactive sessions. To prevent excess backend writes session trackers are no longer created for non-interactive and tbot sessions.


changelog: reduce backend writes incurred by tracking status of non-recorded sessions.

marcoandredinis commented on lib/srv/sess.go:
```diff
@@ -718,7 +718,7 @@ type session struct {
 
 // newSession creates a new session with a given ID within a given context.
-func newSession(ctx context.Context, id rsession.ID, r *SessionRegistry, scx *ServerContext, ch ssh.Channel) (*session, *party, error) {
+func newSession(ctx context.Context, id rsession.ID, r *SessionRegistry, scx *ServerContext, ch ssh.Channel, interactive bool) (*session, *party, error) {
```
💅 We could create a new method here `newInteractiveSession` instead of adding a boolean.

marcoandredinis replied:
I didn't meant to suggest duplicating the logic
Just the entry point: `newTrackedSession/newInteractiveSession` and `newUntrackedSession` (or similar)
And then pull all the logic into a new fn `newBaseSession(..., trackSession bool) ...` (with a godoc indicating that only the above methods should be used.

This is a private method, so we can be more relaxed about it. However, seeing all those `true/false` arguments is somewhat confusing.

rosstimothy committed:
Reduce backend load created from session trackers

Session trackers were originally added to facilitate joining
sessions and enforcing moderation policies. When a session is
created, a new tracker is written to the backend and a background
routine is spawned to periodically update the status of the tracker
until the session is terminated. This can cause a massive amount
of backend activity for a cluster that is spawning large
quantities of sessions per second. While in most cases where
humans are starting the sessions this isn't a problem, any machine
id heavy use cases could trigger backend throttling. Since
non-interactive sessions and sessions started by tbot are not
meant to be joined or moderated, the existence of a session
tracker for them doesn't provide much benefit, especially now that
session recordings are disabled for non-interactive sessions. To
prevent excess backend writes session trackers are no longer
created for non-interactive and tbot sessions.
